activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -68,7 +68,7 @@ def period
 
     # Returns the simultaneous time in <tt>Time.zone</tt>, or the specified zone.
     def in_time_zone(new_zone = ::Time.zone)
-      return self if time_zone == new_zone
+      return self if new_zone == time_zone
```

Comment:
The correct fix here is at `<=>` in `lib/active_support/values/time_zone.rb`.

We should return early if the argument doesn't respond to `utc_offset` and `name`.